<?xml version="1.0" encoding="utf-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="Docutils 0.12: http://docutils.sourceforge.net/" />
<title>pygenomes</title>
<meta name="author" content="FEI YUAN  &lt;fyuan3&#64;uh.edu&gt;" />
<meta name="date" content="2015-07-07" />
<meta name="copyright" content="MIT" />
<style type="text/css">

/*
:Author: David Goodger (goodger@python.org)
:Id: $Id: html4css1.css 7614 2013-02-21 15:55:51Z milde $
:Copyright: This stylesheet has been placed in the public domain.

Default cascading style sheet for the HTML output of Docutils.

See http://docutils.sf.net/docs/howto/html-stylesheets.html for how to
customize this style sheet.
*/

/* used to remove borders from tables and images */
.borderless, table.borderless td, table.borderless th {
  border: 0 }

table.borderless td, table.borderless th {
  /* Override padding for "table.docutils td" with "! important".
     The right padding separates the table cells. */
  padding: 0 0.5em 0 0 ! important }

.first {
  /* Override more specific margin styles with "! important". */
  margin-top: 0 ! important }

.last, .with-subtitle {
  margin-bottom: 0 ! important }

.hidden {
  display: none }

a.toc-backref {
  text-decoration: none ;
  color: black }

blockquote.epigraph {
  margin: 2em 5em ; }

dl.docutils dd {
  margin-bottom: 0.5em }

object[type="image/svg+xml"], object[type="application/x-shockwave-flash"] {
  overflow: hidden;
}

/* Uncomment (and remove this text!) to get bold-faced definition list terms
dl.docutils dt {
  font-weight: bold }
*/

div.abstract {
  margin: 2em 5em }

div.abstract p.topic-title {
  font-weight: bold ;
  text-align: center }

div.admonition, div.attention, div.caution, div.danger, div.error,
div.hint, div.important, div.note, div.tip, div.warning {
  margin: 2em ;
  border: medium outset ;
  padding: 1em }

div.admonition p.admonition-title, div.hint p.admonition-title,
div.important p.admonition-title, div.note p.admonition-title,
div.tip p.admonition-title {
  font-weight: bold ;
  font-family: sans-serif }

div.attention p.admonition-title, div.caution p.admonition-title,
div.danger p.admonition-title, div.error p.admonition-title,
div.warning p.admonition-title, .code .error {
  color: red ;
  font-weight: bold ;
  font-family: sans-serif }

/* Uncomment (and remove this text!) to get reduced vertical space in
   compound paragraphs.
div.compound .compound-first, div.compound .compound-middle {
  margin-bottom: 0.5em }

div.compound .compound-last, div.compound .compound-middle {
  margin-top: 0.5em }
*/

div.dedication {
  margin: 2em 5em ;
  text-align: center ;
  font-style: italic }

div.dedication p.topic-title {
  font-weight: bold ;
  font-style: normal }

div.figure {
  margin-left: 2em ;
  margin-right: 2em }

div.footer, div.header {
  clear: both;
  font-size: smaller }

div.line-block {
  display: block ;
  margin-top: 1em ;
  margin-bottom: 1em }

div.line-block div.line-block {
  margin-top: 0 ;
  margin-bottom: 0 ;
  margin-left: 1.5em }

div.sidebar {
  margin: 0 0 0.5em 1em ;
  border: medium outset ;
  padding: 1em ;
  background-color: #ffffee ;
  width: 40% ;
  float: right ;
  clear: right }

div.sidebar p.rubric {
  font-family: sans-serif ;
  font-size: medium }

div.system-messages {
  margin: 5em }

div.system-messages h1 {
  color: red }

div.system-message {
  border: medium outset ;
  padding: 1em }

div.system-message p.system-message-title {
  color: red ;
  font-weight: bold }

div.topic {
  margin: 2em }

h1.section-subtitle, h2.section-subtitle, h3.section-subtitle,
h4.section-subtitle, h5.section-subtitle, h6.section-subtitle {
  margin-top: 0.4em }

h1.title {
  text-align: center }

h2.subtitle {
  text-align: center }

hr.docutils {
  width: 75% }

img.align-left, .figure.align-left, object.align-left {
  clear: left ;
  float: left ;
  margin-right: 1em }

img.align-right, .figure.align-right, object.align-right {
  clear: right ;
  float: right ;
  margin-left: 1em }

img.align-center, .figure.align-center, object.align-center {
  display: block;
  margin-left: auto;
  margin-right: auto;
}

.align-left {
  text-align: left }

.align-center {
  clear: both ;
  text-align: center }

.align-right {
  text-align: right }

/* reset inner alignment in figures */
div.align-right {
  text-align: inherit }

/* div.align-center * { */
/*   text-align: left } */

ol.simple, ul.simple {
  margin-bottom: 1em }

ol.arabic {
  list-style: decimal }

ol.loweralpha {
  list-style: lower-alpha }

ol.upperalpha {
  list-style: upper-alpha }

ol.lowerroman {
  list-style: lower-roman }

ol.upperroman {
  list-style: upper-roman }

p.attribution {
  text-align: right ;
  margin-left: 50% }

p.caption {
  font-style: italic }

p.credits {
  font-style: italic ;
  font-size: smaller }

p.label {
  white-space: nowrap }

p.rubric {
  font-weight: bold ;
  font-size: larger ;
  color: maroon ;
  text-align: center }

p.sidebar-title {
  font-family: sans-serif ;
  font-weight: bold ;
  font-size: larger }

p.sidebar-subtitle {
  font-family: sans-serif ;
  font-weight: bold }

p.topic-title {
  font-weight: bold }

pre.address {
  margin-bottom: 0 ;
  margin-top: 0 ;
  font: inherit }

pre.literal-block, pre.doctest-block, pre.math, pre.code {
  margin-left: 2em ;
  margin-right: 2em }

pre.code .ln { color: grey; } /* line numbers */
pre.code, code { background-color: #eeeeee }
pre.code .comment, code .comment { color: #5C6576 }
pre.code .keyword, code .keyword { color: #3B0D06; font-weight: bold }
pre.code .literal.string, code .literal.string { color: #0C5404 }
pre.code .name.builtin, code .name.builtin { color: #352B84 }
pre.code .deleted, code .deleted { background-color: #DEB0A1}
pre.code .inserted, code .inserted { background-color: #A3D289}

span.classifier {
  font-family: sans-serif ;
  font-style: oblique }

span.classifier-delimiter {
  font-family: sans-serif ;
  font-weight: bold }

span.interpreted {
  font-family: sans-serif }

span.option {
  white-space: nowrap }

span.pre {
  white-space: pre }

span.problematic {
  color: red }

span.section-subtitle {
  /* font-size relative to parent (h1..h6 element) */
  font-size: 80% }

table.citation {
  border-left: solid 1px gray;
  margin-left: 1px }

table.docinfo {
  margin: 2em 4em }

table.docutils {
  margin-top: 0.5em ;
  margin-bottom: 0.5em }

table.footnote {
  border-left: solid 1px black;
  margin-left: 1px }

table.docutils td, table.docutils th,
table.docinfo td, table.docinfo th {
  padding-left: 0.5em ;
  padding-right: 0.5em ;
  vertical-align: top }

table.docutils th.field-name, table.docinfo th.docinfo-name {
  font-weight: bold ;
  text-align: left ;
  white-space: nowrap ;
  padding-left: 0 }

/* "booktabs" style (no vertical lines) */
table.docutils.booktabs {
  border: 0px;
  border-top: 2px solid;
  border-bottom: 2px solid;
  border-collapse: collapse;
}
table.docutils.booktabs * {
  border: 0px;
}
table.docutils.booktabs th {
  border-bottom: thin solid;
  text-align: left;
}

h1 tt.docutils, h2 tt.docutils, h3 tt.docutils,
h4 tt.docutils, h5 tt.docutils, h6 tt.docutils {
  font-size: 100% }

ul.auto-toc {
  list-style-type: none }

</style>
</head>
<body>
<div class="document" id="pygenomes">
<h1 class="title">pygenomes</h1>
<h2 class="subtitle" id="a-simple-and-excellent-tool-for-downloading-genome-data-from-ucsc-ftp-site">A simple and excellent tool for downloading genome data from UCSC ftp site.</h2>
<table class="docinfo" frame="void" rules="none">
<col class="docinfo-name" />
<col class="docinfo-content" />
<tbody valign="top">
<tr><th class="docinfo-name">Author:</th>
<td>FEI YUAN  &lt;<a class="reference external" href="mailto:fyuan3&#64;uh.edu">fyuan3&#64;uh.edu</a>&gt;</td></tr>
<tr class="field"><th class="docinfo-name">Homepage:</th><td class="field-body"><a class="reference external" href="https://github.com/TinyNecessaryTools/Biology.git">https://github.com/TinyNecessaryTools/Biology.git</a></td>
</tr>
<tr><th class="docinfo-name">Version:</th>
<td>Version 0.1.0</td></tr>
<tr><th class="docinfo-name">Date:</th>
<td>2015-07-07</td></tr>
<tr><th class="docinfo-name">Copyright:</th>
<td>MIT</td></tr>
</tbody>
</table>
<!-- -*- mode: rst -*- -->
<div class="section" id="about">
<h1>About</h1>
<p><cite>pygenomes</cite> is a Python module and also a command-line tool for downloading genome data from UCSC ftp site.</p>
<p>Using this tool you can easily check available taxa or available groups of taxa by simply typing the name of the tool or add -g flag with a group name.</p>
<p>Download genome or chromosome data with this tool also very simple. You can download data by using commom name, scientific name, or even specific assembly name (e.g. <tt class="docutils literal">human</tt>, <tt class="docutils literal">Homo sapiens</tt>, <tt class="docutils literal">hg38</tt>).</p>
<p>If there is available data in the ftp site, you also can download data via interactive mode, which will allow you to choose specific genome assembly.To do this, you only need to assign a True value for assembly, then according to the available list prompted to you, simply input the interested assembly.</p>
</div>
<div class="section" id="features">
<h1>Features</h1>
<ul class="simple">
<li>Install a Python module named <tt class="docutils literal">pygenomes.py</tt></li>
<li>Install a Python command-line script named <tt class="docutils literal">pygenomes</tt></li>
<li>Easily check available taxa and genomes in commend-line</li>
<li>Simply download genomes via common, scientific, or assembly name</li>
<li>Using interactive mode, simply download specific assembly genome</li>
<li>Easily download genome and chromosome in specific file format</li>
<li>Pure Python module and without any third-party dependence</li>
</ul>
</div>
<div class="section" id="examples">
<h1>Examples</h1>
<p>You can use <cite>pygenomes</cite> as a Python module e.g. like in the following
interactive Python session (the function's signature might still change
a bit in the future):</p>
<pre class="literal-block">
&gt;&gt;&gt; from pygenomes import genomes
&gt;&gt;&gt;
# prints out available taxa in group of mammals
&gt;&gt;&gt; genomes(group=&quot;group_name&quot;)                    # i.e. 'mammals'
# download human reference genome (hg19) in .2bit format
&gt;&gt;&gt; genomes(taxa=&quot;taxa&quot;)                           # i.e. 'human'
# interactive mode, ask for inputting and downloading specific assembly
&gt;&gt;&gt; genomes(taxa=&quot;taxa&quot;, assembly=&quot;assembly&quot;)      # i.e. 'cow', '1'
# try to download chromosome data, per fa.gz file per chromosome
&gt;&gt;&gt; genomes(taxa=&quot;taxa&quot;, chrs=&quot;format&quot;)            # i.e. 'cow', '1'
</pre>
<p>In addition, there is a script named <tt class="docutils literal">pygenomes</tt>, which can be used
more easily from the system command-line like this (you can see many
more examples when typing <tt class="docutils literal">pygenomes <span class="pre">-h</span></tt> on the command-line):</p>
<pre class="literal-block">
$ pygenomes -g -1                 # prints out all available taxa

$ pygenomes -g mammal             # prints out all available taxa in group
                                    of mammal
$ pygenomes -t yeast              # download yeast reference genome in
                                    .2bit format
$ pygenomes -t yeast -a 1         # interactive mode, ask for inputing and
                                    downloading specific assembly
$ pygenomes -t cow -f fa.gz       # try to download cow reference genome
                                    in fa.gz format
$ pygenomes -t dog -a 1 -o /tmp   # interactive mode, ask for inputing,
                                    file will be stored in /tmp
$ pygenomes -t cat -c 1           # download cat chromosome data, per
                                    chromosome per fa.gz file
</pre>
</div>
<div class="section" id="installation">
<h1>Installation</h1>
<p>There are three ways to install <cite>pygenomes</cite>, depending on whether you have
module installation tools available on your system or not.</p>
<div class="section" id="using-pip">
<h2>1. Using <cite>pip</cite></h2>
<p>With the <cite>pip</cite> command on your system and a working internet connection, you can install <cite>pygenomes</cite> with only one command in a terminal:</p>
<pre class="literal-block">
$ [sudo] pip install pygenomes
</pre>
<p>If the <cite>pip</cite> command is not available to you and you want to install it before installing <cite>pygenomes</cite>, you might want to go to the
<a class="reference external" href="https://pip.pypa.io/en/stable/installing.html">pip Install homepage</a>
and follow the <a class="reference external" href="https://pip.pypa.io/en/stable/installing.html#install-pip">instructions there</a>.</p>
</div>
<div class="section" id="using-easy-install">
<h2>2. Using <cite>easy_install</cite></h2>
<p>Like pip, with the <cite>easy_install</cite> command on your system and a working internet connection, you can install <cite>pygenomes</cite> also with only one command in a terminal:</p>
<pre class="literal-block">
$ [sudo] easy_install pygenomes
</pre>
<p>If the <cite>easy_install</cite> command is not available to you and you want to
install it before installing <cite>pygenomes</cite>, you might want to go to the
<a class="reference external" href="http://peak.telecommunity.com/DevCenter/EasyInstall">Easy Install homepage</a>
and follow the <a class="reference external" href="http://peak.telecommunity.com/DevCenter/EasyInstall#installing-easy-install">instructions</a>.</p>
</div>
<div class="section" id="manual-installation">
<h2>3. Manual installation</h2>
<p>Alternatively, if you do not have <cite>pip</cite> or <cite>easy_install</cite> and do not want to install one of them first, you can still install the <cite>pygenomes</cite> tarball after downloading the file <tt class="docutils literal"><span class="pre">pygenomes-0.0.1.tar.gz</span></tt> and decompressing it with the following command:</p>
<pre class="literal-block">
$ tar xfz pygenomes-0.0.1.tar.gz
</pre>
<p>Then change into the newly created directory <tt class="docutils literal">pygenomes</tt> and install
<cite>pygenomes</cite> by running the following command:</p>
<pre class="literal-block">
$ [sudo] python setup.py install
</pre>
<p>All three methods will install a Python module file named <tt class="docutils literal">pygenomes.py</tt> in the <tt class="docutils literal"><span class="pre">site-packages</span></tt> subfolder of your Python interpreter and a script
tool named <tt class="docutils literal">pygenomes</tt> in your <tt class="docutils literal">bin</tt> directory, usually in
<tt class="docutils literal">/usr/local/bin</tt> for <a href="#id1"><span class="problematic" id="id2">*</span></a>NIX and Mac OS; a script tool named <tt class="docutils literal">pygenomes.exe</tt> will be created and stored in <tt class="docutils literal">Pythonxx\Scripts</tt> on Windows.</p>
<div class="system-message" id="id1">
<p class="system-message-title">System Message: WARNING/2 (<tt class="docutils">README.txt</tt>, line 122); <em><a href="#id2">backlink</a></em></p>
Inline emphasis start-string without end-string.</div>
</div>
</div>
<div class="section" id="dependencies">
<h1>Dependencies</h1>
<p><cite>Pygenomes</cite> does not depend any third-party module, only need Python standard modules (<cite>os</cite>, <cite>sys</cite>, <cite>subprocess</cite>, <cite>ftplib</cite>, <cite>getopt</cite>). However, I should let you know, when it works, it will use <cite>rsync</cite> first (UCSC recommend download method), if it is not available, it will try to use <cite>wget</cite>, if this also failed, it will download genomes by using ftp (Python standard module <cite>ftplib</cite>, Windows does not have <cite>rsync</cite> and <cite>wget</cite>, it will always use <cite>ftplib</cite>). So, there is no need for you to worry about its dependencies, the only thing you need to do is install it and then enjoy it.</p>
</div>
<div class="section" id="bug-reports">
<h1>Bug reports</h1>
<p>Please report bugs and patches to FEI YUAN &lt;<a class="reference external" href="mailto:fyuan4&#64;uh.edu">fyuan4&#64;uh.edu</a>&gt;
Don't forget to include information about the operating system and Python versions being used.</p>
</div>
</div>
</body>
</html>
